### ISSUE TITLE
Wrapping a ContentCollection in a ContentComponent breaks backend rendering of first element

### ISSUE THREAD

**Issue by klfman** [neos/neos-development-collection#2958]
### Description

When wrapping a `Neos.Neos:ContentCollection` with a `Neos.Neos:ContentComponent`, the rendering of the inline ui in the neos backend breaks when adding the first element to the content collection. When wrapping it with a `Neos.Fusion:Component` instead it works without any problems.

The visible effect of this is that the content is rendered **next** to the content collection and **not into** it. The content collection still shows with the empty overlay.
(I cannot take a screenshot right now with my current setup, if needed I could add one later.)

When reloading the page, however, as long as there is at least one item in the content collection, the problem is gone without changing the content.

Further drilling into the problem shows: A duplicate contentElementWrapping creates this problem (see the files with the html markup extracted with my browsers dev tools that I attached below).

### Steps to Reproduce

Using these definitions of a simple section
```yaml
'Vendor.Site:Section':
  superTypes:
    'Neos.Neos:Content': true
    'Neos.Neos:ContentCollection': true
  ui:
    label: 'Section'
  constraints:
    nodeTypes:
      'Vendor.Site:Text': true
```

with the fusion prototype:
```
prototype(Vendor.Site:Section) < prototype(Neos.Neos:ContentComponent) {
    renderer = Neos.Neos:ContentCollection {
        nodePath = '.'
    }
}
```

And a simple text node type:
```yaml
'Vendor.Site:Text':
  superTypes:
    Neos.Neos:Content: true
  ui:
    label: Text
    icon: icon-file-text
  properties:
    text:
      type: string
      defaultValue: 'Text example'
      ui:
        inlineEditable: true
```

with the fusion prototype:
```
prototype(Vendor.Site:Text) < prototype(Neos.Neos:ContentComponent) {
    renderer = Neos.Neos:Editable {
        property = 'text'
    }
}
```

Adding the Text to the section breaks the backend UI, showing the empty content collection overlay next to the added text.
When the page is then reloaded, the rendering is correct, showing the text within the content collection.

**The generated html markup extracted with my browsers dev tools confirms the visual effect:**

[before_reload.txt](https://github.com/neos/neos-development-collection/files/4458642/before_reload.txt)
[after_reload.txt](https://github.com/neos/neos-development-collection/files/4458643/after_reload.txt)

## Observation
if the section inherits from Neos.Fusion:Component the error does not happen. So the duplicate contentElementWrapping applied by both the `ContentCollection` and the `ContentComponent` must be the problem here.

To confirm this, change the line above to this one and it works:
`prototype(Vendor.Site:Section) < prototype(Neos.Fusion:Component)`

### Affected Versions

Neos: 4.3.12
Flow: 5.3.14

**Comment by jonnitto** [comment]
Hi @klfman 
This feature was added in Neos 5.2
Did you read this article from @Sebobo? https://mind-the-seb.de/blog/preventing-deep-content-nesting-in-neos-cms

**Comment by klfman** [comment]
Hey @jonnitto,

thanks for pointing me to the article (which describes a valid workaround). I also had a look at the PR, I'm fine with using the proposed workaround.

This issue is effectively solved by https://github.com/neos/neos-ui/pull/2609 when I upgrade to Neos 5.2.

However, I feel like the default behavior of ContentCollections introduced with that pull request should be backported to 4.3, as the current behavior feels broken, if I, as a developer, just stick to the usual practise of using ContentComponents.

**Comment by Sebobo** [comment]
The problem is that the backport has to happen for Neos and the UI. 
I don't fully see it as a bug and more as a feature. And others see it as a bug. 
I don't have a problem when someone makes a backport and people agree to merge it as a bugfix. But I personally currently don't want to do the work to do the backport.

**Comment by klfman** [comment]
I'm fine with not backporting, as there is a workaround available that works (thanks for the blogpost @Sebobo !).

I'm closing this as backporting or not will not depend on this issue and it's known and fixed in Neos 5.2.
